// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTConsensusEnums";
option optimize_for = LITE_RUNTIME;

enum ConsensusType {
    CONSENSUSTYPE_ERROR = 0;
    CONSENSUSTYPE_SERVER = 1;
    CONSENSUSTYPE_CLIENT = 2;
    CONSENSUSTYPE_PEER = 3;
}

enum DeliveryState {
    DELIVERTYSTATE_ERROR = 0;
    DELIVERTYSTATE_IDLE = 1;
    DELIVERTYSTATE_PENDINGSEND = 2;
    DELIVERTYSTATE_NEEDNYMBOX = 3;
    DELIVERTYSTATE_NEEDBOXITEMS = 4;
    DELIVERTYSTATE_NEEDPROCESSNYMBOX = 5;
}

enum LastReplyStatus {
    LASTREPLYSTATUS_INVALID = 0;
    LASTREPLYSTATUS_NONE = 1;
    LASTREPLYSTATUS_MESSAGESUCCESS = 2;
    LASTREPLYSTATUS_MESSAGEFAILED = 3;
    LASTREPLYSTATUS_UNKNOWN = 4;
    LASTREPLYSTATUS_NOTSENT = 5;
}
